### PR TITLE
Kruskal Algorithm Implementation in python

### LINKED ISSUE #2200: Kruskal Algorithm
- [ ] C - @coderprasukj
- [x] C++
- [ ] C# - @chandrikadeb7
- [ ] CoffeeScript
- [ ] Dart - @Swarnimashukla
- [ ] Go
- [x] Java
- [ ] JS - @shreyakapoor08
- [ ] Kotlin - @garimasingh128
- [ ] Php - @prince-09
- [x] Python - @Manish-cloud
- [x] Readme
- [ ] Ruby 
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Manish-cloud**
Used disjoint set data structure for Algorithm implementation.

Fixes #2200  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added implementation of Kruskal Algorithm
- 

#### Languages Used:
- Python 3.6
- 

#### Files Added:
- Kruskal_Algorithm.py
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Manish-cloud** [comment]
@singh-shreya6 Kindly review the PR . Adding Python Implementation for Kruskal Algorithm.

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+5 c e
+6 d e
+
+Minimun Weight :  11
+('a', 'c') => 1
+('c', 'b') => 2
+('b', 'd') => 3
+('c', 'e') => 5
+
+"""
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:105]: Please add blank line at end of file.

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    numberOfEdges=int(input("Enter number of edges in graph : "))
+    print("Note :=> Enter graph data in following format weight vertex1 vertex2")
+    for _ in range(numberOfEdges):
+        weights,vertex1,vertex2 = input().split()
+        weights=int(weights)
+        graphData.append((weights,vertex1,vertex2))
+    
+    kruskalAlgorithm(vertices,graphData)
+    
+
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:73]: 1-2 blank lines are enough. Please remove others.

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        vertices.append(vertex)
+
+    numberOfEdges=int(input("Enter number of edges in graph : "))
+    print("Note :=> Enter graph data in following format weight vertex1 vertex2")
+    for _ in range(numberOfEdges):
+        weights,vertex1,vertex2 = input().split()
+        weights=int(weights)
+        graphData.append((weights,vertex1,vertex2))
+    
+    kruskalAlgorithm(vertices,graphData)
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:71]: Please add space after comma

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    for _ in range(numberOfVertices):
+        vertex = input()
+        vertices.append(vertex)
+
+    numberOfEdges=int(input("Enter number of edges in graph : "))
+    print("Note :=> Enter graph data in following format weight vertex1 vertex2")
+    for _ in range(numberOfEdges):
+        weights,vertex1,vertex2 = input().split()
+        weights=int(weights)
+        graphData.append((weights,vertex1,vertex2))
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:69]: Space after comma

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    print("Enter vertices")
+    for _ in range(numberOfVertices):
+        vertex = input()
+        vertices.append(vertex)
+
+    numberOfEdges=int(input("Enter number of edges in graph : "))
+    print("Note :=> Enter graph data in following format weight vertex1 vertex2")
+    for _ in range(numberOfEdges):
+        weights,vertex1,vertex2 = input().split()
+        weights=int(weights)
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:68]: Space around =

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        # when length of disjoint set becomes 1 we can break loop
+        
+        if (len(adisjointSet.sets)==1):
+            break
+        
+        """        
+    print("Minimun Weight : ",minimumWeight)    
+    # To display in which order edges are added    
+    for path in MST:
+        print((path[1],path[2]),"=>",path[0])
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:54]: Space after comma

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    for weight,vertex1,vertex2 in graphData:
+        if (adisjointSet.findSet(vertex1) != adisjointSet.findSet(vertex2)):
+            minimumWeight+=weight
+            MST.append((weight,vertex1,vertex2))
+            adisjointSet.mergeSets(vertex1,vertex2)
+        """
+        # To improve efficiency
+        # when length of disjoint set becomes 1 we can break loop
+        
+        if (len(adisjointSet.sets)==1):
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:47]: Please add Space aound operator at all places on this file.  =,  ==

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+    def mergeSets(self, val1, val2):
+        set1 = self.findSet(val1)
+        set2 = self.findSet(val2)
+        if set1 != set2:
+            set1.extend(set2)
+            self.sets.remove(set2)
+
+def kruskalAlgorithm(vertices,graphData):
+    MST=[]
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:29]: Space around = in both the lines here.

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 9]
```diff
@@ -0,0 +1,105 @@
+#Implementation of Kruskal Algorithm using Disjoint 
+
+graphData=[]
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:3]: Space around =

**Comment by Manish-cloud** [comment]
@singh-shreya6 Please review my PR. Requested changes done in code.

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+        if (adisjointSet.findSet(vertex1) != adisjointSet.findSet(vertex2)):
+            minimumWeight += weight
+            MST.append((weight, vertex1, vertex2))
+            adisjointSet.mergeSets(vertex1, vertex2)
+        # To improve efficiency
+        # when length of disjoint set becomes 1 we can break loop
+        if (len(adisjointSet.sets) == 1):
+            break
+    
+    print("Minimun Weight : ", minimumWeight)
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:48]: Please fix typo in word Minimum

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+        if (len(adisjointSet.sets) == 1):
+            break
+    
+    print("Minimun Weight : ", minimumWeight)    
+    # To display in which order edges are added    
+    for path in MST:
+        print((path[1], path[2]), "=>", path[0])
+    
+if __name__ == "__main__":
+    numberOfVertices=int(input("Enter number of vertices in graph : "))
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:54]: Space around =

File: Kruskal_Algorithm/Kruskal_Algorithm.py [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+Enter number of edges in graph : 7
+Note :=> Enter graph data in following format weight vertex1 vertex2
+1 a c
+4 a b
+2 c b
+3 b d
+7 c d
+5 c e
+6 d e
+Minimun Weight :  11
```
Comment (singh-shreya6) [inline Kruskal_Algorithm/Kruskal_Algorithm.py:88]: Fix typo in word Minimum

**Comment by singh-shreya6** [comment]
Please learn to squash commits soon and make the required change.

**Comment by Manish-cloud** [comment]
@singh-shreya6 Please review my PR.
Completed squashing and requested changes.

**Comment by singh-shreya6** [comment]
Good work @Manish-cloud.

**Pull request merged**